run-tests.sh

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+cd "$(dirname "$0")"
+
+port=9000
+
+print_usage() {
+  cat <<EOF
+usage: $0 [IMAGE...]
+
+examples:
+       $0 "sonar-scanner-cli:4.1"
+EOF
+}
+
+info() {
+  echo "[info] $@"
+}
+
+warn() {
+  echo "[warn] $@" >&2
+}
+
+fatal() {
+  echo "[error] $@" >&2
+  exit 1
+}
+
+require() {
+  local prog missing=()
+  for prog; do
+    if ! type "$prog" &>/dev/null; then
+      missing+=("$prog")
+    fi
+  done
+
+  [[ ${#missing[@]} == 0 ]] || fatal "could not find reqired programs on the path: ${missing[@]}"
+}
+
+wait_for_sonarqube() {
+  local i web_up=no sonarqube_up=no
+
+  for ((i = 0; i < 10; i++)); do
+    info "waiting for web server to start ..."
+    if curl -sI localhost:$port | grep '^HTTP/.* 200'; then
+      web_up=yes
+      break
+    fi
+    sleep 5
+  done
+
+  [[ $web_up == yes ]] || return 1
+
+  for ((i = 0; i < 20; i++)); do
+    info "waiting for sonarqube to be ready ..."
+    if curl -s localhost:$port/api/system/status | grep '"status":"UP"'; then
+      sonarqube_up=yes
+      break
+    fi
+    sleep 10
+  done
+
+  [[ "$sonarqube_up" == yes ]]
+}
+
+test_scanner() {
```

Comment:
similar comment as below:
This works on an ephemeral machine but I wonder whether the clone shouldn't rather be done in a temp directory erased at the end of the script? 

I agree, if we do so, providing an option to not erase the directory to allow investigation becomes quite important and makes the idea of erasing the temp directory quite challengeable...

Reply:
Done.